datasource/opennms-direct/src/main/resources/inventory.groovy

```diff
@@ -0,0 +1,180 @@
+        if (!Strings.isNullOrEmpty(alarm.getManagedObjectType()) && !Strings.isNullOrEmpty(alarm.getManagedObjectInstance())) {
+            ManagedObjectType type;
+            try {
+                type = ManagedObjectType.fromName(alarm.getManagedObjectType());
+            } catch (NoSuchElementException nse) {
+                log.error("Found unsupported type: {} with id: {}. Skipping.", alarm.getManagedObjectType(), alarm.getManagedObjectInstance());
+                return;
+            }
+
+            Set<ManagedObjectType> alreadyScoped = new HashSet<>(Arrays.asList(
+                    ManagedObjectType.Node,
+                    ManagedObjectType.SnmpInterfaceLink,
+                    ManagedObjectType.EntPhysicalEntity,
+                    ManagedObjectType.BgpPeer,
+                    ManagedObjectType.VpnTunnel
+                    ));
+
+            if (!alreadyScoped.contains(type)) {
+                alarmBuilder.setInventoryObjectType(type.getName());
+                alarmBuilder.setInventoryObjectId(String.format("%s:%s", toNodeCriteria(alarm.getNode()), alarm.getManagedObjectInstance()));
+            }
+        }
+
+        if ((alarm.getManagedObjectType() == null || alarm.getManagedObjectInstance() == null) && alarm.getNode() != null) {
+            alarmBuilder.setInventoryObjectType(ManagedObjectType.Node.getName());
+            alarmBuilder.setInventoryObjectId(toNodeCriteria(alarm.getNode()));
+        }
+
+    }
+
+    static ImmutableInventoryObject toInventoryObject(SnmpInterface snmpInterface, InventoryObject parent) {
+        return ImmutableInventoryObject.newBuilder()
+                .setType(ManagedObjectType.SnmpInterface.getName())
+                .setId(parent.getId() + ":" + snmpInterface.getIfIndex())
+                .setFriendlyName(snmpInterface.getIfDescr())
+                .setParentType(parent.getType())
+                .setParentId(parent.getId())
+                .build();
+    }
+
+    static String toNodeCriteria(Alarm alarm) {
+        log.info("alarmToNodeCriteria {}", alarm);
+        Node node = alarm.getNode();
+        if (node != null) {
+            return toNodeCriteria(node);
+        }
+        return toNodeCriteria(null, null, alarm.getId());
+    }
+
+    static String toNodeCriteria(Node node) {
+        log.info("nodeToCriteria {}", node);
+        return toNodeCriteria(node.getForeignSource(), node.getForeignId(), node.getId());
+    }
+
+    static String toNodeCriteria(String foreignSource, String foreignId, int id) {
+        log.info("toNodeCriteria: {} {} {}", foreignSource, foreignId, id);
+        if (!Strings.isNullOrEmpty(foreignSource) && !Strings.isNullOrEmpty(foreignId)) {
+            return foreignSource + ":" + foreignId;
+        } else {
+            return Integer.valueOf(id).toString();
+        }
+    }
+}
+
+def Collection<InventoryObject> alarmToInventory(Alarm alarm) {
+    return InventoryFactory.alarmToInventory(alarm);
```

Comment:
Do these method calls need the class qualifier?

Also could you just add the `def` to the other method to remove these proxies?

Reply:
The global method defs are what are exposed to the ScriptEngine.
A class is required in order to support logging using the @Slf4j annotation.
That's how the script ended up with the proxies that just call through to the class methods.
I think the qualifiers just make it more clear, since they may be looked at on their own.

Reply:
Ok, not that familiar with groovy land so was just wondering.